### PR TITLE
Platforms, jumping, saves, and enemies. TLDR in changelog

### PR DESCRIPTION
**Pull request by MenacingManatee**
TLDR/CHANGELOG:
* Cooldown on jumps added
* Agility now affects jump cooldown
* Minor bugfixes to saving
* Visual indicator added for jump cooldown
* Added an override to jump function if player needs to keep jumping regardless of cooldown. Mainly so they don't jump on an enemy and immediately fall through and take damage.
* Added placeholder rat sprite
* Created rat class
* Added placeholder platform sprite
* Created platform class
* Platforms tested with gravity of player. May need some tweaking, but I think I've got it working.
* Rope class added
* Player can slowly climb a rope
* Player can press down to move through platforms. May have unintended bugs with real ground later, needs further debugging

KNOWN BUGS
* Slowfall isn't working. Apparently is never true.
* The very top of a rope is a bit glitchy, causes the player to spaz out while holding up at the top
* Jumping at the bottom of a rope sometimes causes a super jump
* Forgot to remove debug text, expect a lot of spam until I get rid of that (i.e., ~60 lines per second lol)

### CODE REVIEW

**Comment by MenacingManatee** [comment]
@GabrielAbdul changes to save are so save_1 isn't overwritten every time a new game is made, and all saves aren't overwritten every time a new save is saved

**Comment by MenacingManatee** [comment]
Might need to play with jump a bit more to give them some forward momentum when jumping. It'll be hard to clear larger gaps with the current jump distance, but I suppose this is more a puzzler, not a platformer

**Comment by MenacingManatee** [comment]
Could make it so agility affects cooldowns and base walking speed, but strength affects jump height and speed while jumping. Could be difficult to implement, but not impossible

**Comment by MenacingManatee** [comment]
Glow could also step in as damage reduction or heal-over-time. Unless there's a time limit, a heal cap per level, or its almost uselessly slow, HOT might be game breaking though, and with how low the damage is, percent health blocking would be useless and flat blocking would be broken. Maybe a shield that gets depleted before health, and is refilled each level? Depending on how easy it is to accumulate glow, that could work. Idk, just marking down my idle thoughts so I can look back for inspiration later and maybe get some feedback if one catches an eye lol

**Comment by MenacingManatee** [comment]
Maybe glow could serve as a currency of sorts, that lets you spec into upgrades. I remember you mentioned an idea like that back when we first started gabe. You could purchase say, 1 dmg flat blocking for 5 glow, and that wouldn't be terrible, especially if we had minimum damage or a cap on how much you could buy, or both. Would also encourage players to get a lot of glow, which if it can be earned by killing would encourage that, but we could also do it like Bioshock where you get less immediately by not killing, but more overall through some mechanic rewarding the player for taking the 'good' route

**Comment by MenacingManatee** [comment]
Platform will need some tweaks before the crate is added as a platform. Mainly to make the collision for gravity based wholly on the top. Might warrant an extra class if we want to have collision on the sides too, say for pushing, but there might be a simpler way
